cognite/neat/rules/importers/_spreadsheet2rules.py

```diff
@@ -77,43 +171,38 @@ def to_rules(
         issue_list = IssueList(title=f"'{self.filepath.name}'")
 
         if not self.filepath.exists():
-            issue_list.append(cognite.neat.rules.issues.spreadsheet_file.SpreadsheetNotFoundError(self.filepath))
-            if error_handling == "raise":
-                raise issue_list.as_errors() from None
-            return None, issue_list
+            issue_list.append(issues.spreadsheet_file.SpreadsheetNotFoundError(self.filepath))
+            return self._return_or_raise(issue_list, error_handling)
 
+        user_rules: Rules | None = None
         if not is_reference:
-            user_rules, issue_list = self._to_single_rules(issue_list, RulesCategory.user, error_handling)
-        else:
-            user_rules = None
-
-        if issue_list:
-            if error_handling == "raise":
-                raise issue_list.as_errors()
-            return None, issue_list
+            user_rules = SpreadsheetReader(issue_list, is_reference=False).read(self.filepath)
+            if issue_list.has_errors:
+                return self._return_or_raise(issue_list, error_handling)
 
+        reference_rules: Rules | None = None
         if is_reference or (
             user_rules
             and user_rules.metadata.role != RoleTypes.domain_expert
             and cast(DMSRules | InformationRules, user_rules).metadata.schema_ == SchemaCompleteness.extended
         ):
-            reference_rules, issue_list = self._to_single_rules(issue_list, RulesCategory.reference, error_handling)
-        else:
-            reference_rules = None
+            reference_rules = SpreadsheetReader(issue_list, is_reference=True).read(self.filepath)
+            if issue_list.has_errors:
+                return self._return_or_raise(issue_list, error_handling)
 
-        if issue_list:
-            if error_handling == "raise":
-                raise issue_list.as_errors()
-            return None, issue_list
+        if user_rules and reference_rules and user_rules.metadata.role != reference_rules.metadata.role:
+            issue_list.append(issues.spreadsheet_file.RoleMismatchError(self.filepath))
+            return self._return_or_raise(issue_list, error_handling)
 
         if user_rules and reference_rules:
-            if user_rules.metadata.role != reference_rules.metadata.role:
-                issue_list.append(cognite.neat.rules.issues.spreadsheet_file.RoleMismatchError(self.filepath))
-                if error_handling == "raise":
-                    raise issue_list.as_errors()
-                return None, issue_list
-
-        rules = cast(DomainRules | InformationRules | DMSRules, user_rules or reference_rules)
+            rules = user_rules
+            rules.reference = reference_rules
+        elif user_rules:
+            rules = user_rules
+        elif reference_rules:
+            rules = reference_rules
+        else:
+            raise ValueError("No rules found.")
```

Comment:
I added this part, as I think we want to pass on the reference rules with the user rules if they are both present. However, this implementation does not trigger validation, so we might have to rethink this a bit. But waiting to get feedback on the suggestion in this PR.